src/any_llm/providers/azure/azure.py

```diff
@@ -125,6 +202,29 @@ def completion(
 
         return _convert_response(response)
 
+    async def aembedding(
+        self,
+        model: str,
+        inputs: str | list[str],
+        **kwargs: Any,
+    ) -> CreateEmbeddingResponse:
+        """Create embeddings using Azure AI Inference SDK."""
+        client: aio.EmbeddingsClient = self._create_embeddings_client_async()
+
+        input_list: list[str]
+        if isinstance(inputs, str):
+            input_list = [inputs]
+        else:
+            input_list = inputs
+
+        response: EmbeddingsResult = await client.embed(
+            model=model,
+            input=input_list,
+            **kwargs,
+        )
```

Comment:
Maybe a bit more pythonic
```suggestion
        response: EmbeddingsResult = await client.embed(
            model=model,
            input=inputs if isinstance(inputs, list) else [inputs],
            **kwargs,
        )
```